modules/SofaGraphComponent/SceneCheckerVisitor.cpp

```diff
@@ -0,0 +1,85 @@
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License *
+* for more details.                                                           *
+*                                                                             *
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+*******************************************************************************
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+#include "SceneCheckerVisitor.h"
+#include "RequiredPlugin.h"
+#include <sofa/core/ObjectFactory.h>
+#include <sofa/helper/system/PluginManager.h>
+
+namespace sofa
+{
+namespace simulation
+{
+using sofa::component::misc::RequiredPlugin ;
+using sofa::core::ObjectFactory ;
+using sofa::core::ExecParams ;
+using sofa::helper::system::PluginRepository ;
+using sofa::helper::system::PluginManager ;
+
+SceneCheckerVisitor::SceneCheckerVisitor(const ExecParams* params) : Visitor(params)
+{
+}
+
+SceneCheckerVisitor::~SceneCheckerVisitor()
+{
+}
+
+void SceneCheckerVisitor::validate(Node* node)
+{
+    helper::vector< RequiredPlugin* > plugins ;
+    node->getTreeObjects< RequiredPlugin >(&plugins) ;
+
+    for(auto& plugin : plugins)
+        m_pluginsrequired[plugin->getName()] = true ;
+
+    execute(node) ;
+}
+
+Visitor::Result SceneCheckerVisitor::processNodeTopDown(Node* node)
+{
+    for (auto& object : node->object )
+    {
+        ObjectFactory::ClassEntry entry = ObjectFactory::getInstance()->getEntry(object->getClassName());
+        if(!entry.creatorMap.empty())
+        {
+            ObjectFactory::CreatorMap::iterator it = entry.creatorMap.find(object->getTemplateName());
+            if(entry.creatorMap.end() != it && *it->second->getTarget()){
+                std::string pluginName = it->second->getTarget() ;
+                std::string path = PluginManager::getInstance().findPlugin(pluginName) ;
+                if( PluginManager::getInstance().pluginIsLoaded(path)
+                    && m_pluginsrequired.find(pluginName) == m_pluginsrequired.end() )
+                {
+                    msg_warning("SceneChecker") << "This scene is using component '" << object->getClassName() << "'. " << msgendl
+                                                << "This component is part of the '" << pluginName << "' plugin but there is no RequiredPlugin name='" << pluginName << "' directive in your scene." << msgendl
+                                                << "Your scene may not work on a sofa environment that does not have a pre-loaded the plugin." << msgendl
```

Comment:
Hi @guparan, thanks for your interest :) 

Actually the plugin warns a user that it is loading a scene the requires a plugin but that as it have this plugin pre-loaded this scene will only work on his own sofa installation. 

With the sentence you suggest it give the impression that what is "missing" is to pre-loaded a plugin. This is not true... pre-loading a plugin will not remove the message... because the only correct and general solution is to have all the RequiredPlugin in the scenes to make them "self loadable".  

Do you see the point ?